physical/raft/raft.go

```diff
@@ -916,9 +916,9 @@ func (b *RaftBackend) SetupCluster(ctx context.Context, opts SetupOpts) error {
 			return true
 		default:
 			// Concrete type checks
-			switch cl.(type) {
+			switch listener := cl.(type) {
```

Comment:
All of this could also become:

```go
	listenerIsNil := func(cl cluster.ClusterHook) bool {
		if opts.ClusterListener == nil {
			return true
		}
		if listener, ok := cl.(*cluster.Listener); ok {
			return listener == nil
		}
		return false
	}
```

Reply:
I'll keep this as is for now, to not accidentally introduce any functional changes.